src/fusion_trees/fusiontree.jl

```diff
@@ -0,0 +1,326 @@
+#
+# --------------- misc  ---------------
+function to_tuple(f::SectorFusionTree)
+  return (
+    leaves(f)...,
+    arrows(f)...,
+    root_sector(f),
+    branch_sectors(f)...,
+    outer_multiplicity_indices(f)...,
+  )
+end
+
+# --------------- SectorProduct helper functions  ---------------
+function outer_multiplicity_kron(
+  sec1, sec2, fused, outer_multiplicity1, outer_multiplicity2
+)
+  n = nsymbol(sec1, sec2, fused)
+  linear_inds = LinearIndices((n, outer_multiplicity2))
+  return linear_inds[outer_multiplicity1, outer_multiplicity2]
+end
+
+# TODO move to GradedUnitRanges
+function nsymbol(s1::AbstractSector, s2::AbstractSector, s3::AbstractSector)
+  full_space = fusion_product(s1, s2)
+  x = findfirst(==(s3), blocklabels(full_space))
+  isnothing(x) && return 0  # OR labelled(0, s3)?
+  return Int(blocklengths(full_space)[x])
+end
+
+function outer_multiplicity_split(
+  sec1::S, sec2::S, fused::S, outer_mult_index::Integer
+) where {S<:SectorProduct}
+  args1 = arguments(sec1)
+  args2 = arguments(sec2)
+  args12 = arguments(fused)
+  nsymbols = Tuple(map(nsymbol, args1, args2, args12))  # CartesianIndices requires explicit Tuple
+  return Tuple(CartesianIndices(nsymbols)[outer_mult_index])
+end
+
+# --------------- Build trees  ---------------
+# zero leg: need S to get sector type information
+function SectorFusionTree{S}() where {S<:AbstractSector}
+  return SectorFusionTree((), (), trivial(S), (), ())
+end
+function SectorFusionTree{S}(::Tuple{}, ::Tuple{}) where {S<:AbstractSector}
+  return SectorFusionTree((), (), trivial(S), (), ())
+end
+
+# one leg
+function SectorFusionTree(sect::AbstractSector, arrow::Bool)
+  return SectorFusionTree((sect,), (arrow,), sect, (), ())
+end
+
+function braid_tuples(t1::Tuple{Vararg{Any,N}}, t2::Tuple{Vararg{Any,N}}) where {N}
+  t12 = (t1, t2)
+  nested = ntuple(i -> getindex.(t12, i), N)
+  return flatten_tuples(nested)
+end
+
+function grow_tree(
+  parent_tree::SectorFusionTree,
+  branch_sector::AbstractSector,
+  level_arrow::Bool,
+  child_root_sector,
+  outer_mult,
+)
+  child_leaves = (leaves(parent_tree)..., branch_sector)
+  child_arrows = (arrows(parent_tree)..., level_arrow)
+  child_branch_sectors = (branch_sectors(parent_tree)..., root_sector(parent_tree))
+  child_outer_mul = (outer_multiplicity_indices(parent_tree)..., outer_mult)
+  return SectorFusionTree(
+    child_leaves, child_arrows, child_root_sector, child_branch_sectors, child_outer_mul
+  )
+end
+
+function grow_tree(
+  parent_tree::SectorFusionTree, branch_sector::AbstractSector, level_arrow::Bool
+)
+  new_space = fusion_product(root_sector(parent_tree), branch_sector)
+  return mapreduce(vcat, zip(blocklabels(new_space), blocklengths(new_space))) do (la, n)
+    return [
+      grow_tree(parent_tree, branch_sector, level_arrow, la, outer_mult) for
+      outer_mult in 1:n
+    ]
+  end
+end
+
+function build_trees(old_trees::Vector, sectors_to_fuse::Tuple, arrows_to_fuse::Tuple)
```

Comment:
I guess this one is more obvious since I see that it is expanding thick fusion trees to thin/sector fusion trees, but maybe we can also have a more explicit name like `expand_to_sector_fusion_trees`.